diff --git a/src/samples/Samples/5_Domain_Specific/BlackScholes/BlackScholes.cu.hip b/src/samples/Samples/5_Domain_Specific/BlackScholes/BlackScholes.cu.hip
index c066a41..62f54ed 100644
--- a/src/samples/Samples/5_Domain_Specific/BlackScholes/BlackScholes.cu.hip
+++ b/src/samples/Samples/5_Domain_Specific/BlackScholes/BlackScholes.cu.hip
@@ -35,7 +35,7 @@
 #include <hip/hip_runtime.h>
 #include <helper_functions.h>  // helper functions for string parsing
 #include <helper_cuda.h>  // helper functions CUDA error checking and initialization
-
+#include "HIPCHECK.h"
 ////////////////////////////////////////////////////////////////////////////////
 // Process an array of optN options on CPU
 ////////////////////////////////////////////////////////////////////////////////
